app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tab/Addresses.php

```diff
@@ -127,10 +127,25 @@ public function initForm()
             Mage::helper('adminhtml/addresses')
                 ->processStreetAttribute($attributes['street']);
         }
+
+        $customerStoreId = null;
+        if ($customer->getId()) {
+            $customerStoreId = Mage::app()->getWebsite($customer->getWebsiteId())->getDefaultStore()->getId();
+        }
```

Comment:
I see that the code block is moved from line 163. But if a website has more than one store, with mutually exclusive country list, e.g., the website's default store is store1, which has only France, store2 has only Germany, etc. Wouldn't non-French customers get incorrect country list? Is this case tested?

```suggestion
        $customerStoreId = $customer->getStoreId(); // null if new customer
```

Not tested, but I think the above should work for `suffix` and `prefix` later on, lines 180 and 194.